File: 0169-majority-element/0169-majority-element.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        """
        most bin 은 n/2 만큼
        n^2 까진 한도
        정렬 후 그 빈도수 체크로 나아가면 될 듯
        max 값을 유지하면서 가는 것.(value, 빈도)
        """
        max_bin = dict()

        for i, k in enumerate(nums):
            if k in max_bin:
                max_bin[k] += 1
            else:
                max_bin[k] = 1
        
        ans = max(max_bin, key=max_bin.get)
        return ans


        

File: 0169-majority-element/0169-majority-element.cs
public class Solution {
    public int MajorityElement(int[] nums) {
        int candidate = 0;
        int count = 0;

        foreach (int num in nums){
            if (count == 0) {
                candidate = num;
            }
            _ = (num == candidate) ? count++ : count--;
        }

        return candidate;
    }
}